packages/big-design/src/components/Checkbox/Checkbox.tsx

```diff
@@ -18,102 +19,92 @@ interface PrivateProps {
-          disabled={disabled}
-          hidden={hiddenLabel}
-          htmlFor={htmlFor}
-          aria-hidden={disabled}
-          id={this.labelUniqueId}
-        >
+        <CheckboxLabel disabled={disabled} hidden={hiddenLabel} htmlFor={id} aria-hidden={disabled} id={labelId}>
           {label}
-        </StyledLabel>
+        </CheckboxLabel>
       );
     }
 
-    if (React.isValidElement(label) && label.type === Checkbox.Label) {
+    if (React.isValidElement(label) && label.type === CheckboxLabel) {
       return React.cloneElement(label as React.ReactElement<React.LabelHTMLAttributes<HTMLLabelElement>>, {
         hidden: hiddenLabel,
-        htmlFor,
-        id: this.labelUniqueId,
+        htmlFor: id,
+        id: labelId,
       });
     }
 
-    return null;
-  }
-}
+    if (!label) {
+      return null;
+    }
 
-const CheckboxWithForwardedRef = React.forwardRef<HTMLInputElement, CheckboxProps>(
-  ({ className, style, ...props }, ref) => <RawCheckbox {...props} forwardedRef={ref} />,
+    warning('label must be either a string or a CheckboxLabel component.');
+  }, [disabled, hiddenLabel, id, label, labelId]);
+
+  return (
+    <CheckboxContainer className={className} style={style}>
+      <HiddenCheckbox
+        type="checkbox"
+        checked={checked}
+        id={id}
+        disabled={disabled}
+        {...props}
+        aria-labelledby={labelId}
+        ref={checkbox => {
+          if (checkbox && typeof isIndeterminate === 'boolean') {
+            checkbox.indeterminate = !checked && isIndeterminate;
+          }
+
+          if (forwardedRef) {
+            if (typeof forwardedRef === 'function') {
+              forwardedRef(checkbox);
+            } else {
+              // RefObject.current is readonly in DefinitelyTyped but in practice you
+              // can still write to it.
+              // See https://github.com/DefinitelyTyped/DefinitelyTyped/issues/31065
+              // @ts-ignore
+              forwardedRef.current = checkbox;
+            }
+          }
+        }}
+      />
+
+      <StyledCheckbox
+        disabled={disabled}
+        isIndeterminate={isIndeterminate}
+        checked={checked}
+        htmlFor={id}
+        aria-hidden={true}
+      >
+        {!checked && isIndeterminate ? <RemoveIcon /> : <CheckIcon />}
+      </StyledCheckbox>
+      {renderedLabel}
+    </CheckboxContainer>
+  );
+};
+
+export const Checkbox = typedMemo(
+  forwardRef<HTMLInputElement, CheckboxProps>(({ className, style, ...props }, ref) => (
+    <RawCheckbox {...props} forwardedRef={ref} />
+  )),
 );
 
-export const Checkbox = hoistNonReactStatics(CheckboxWithForwardedRef, RawCheckbox);
-
-export const StyleableCheckbox = React.forwardRef<HTMLInputElement, CheckboxProps>((props, ref) => (
-  <RawCheckbox {...props} forwardedRef={ref} />
-));
-
-Checkbox.displayName = 'Checkbox';
-StyleableCheckbox.displayName = 'StyleableCheckbox';
+export const StyleableCheckbox = typedMemo(
```

Comment:
Since we aren't using `StyleableCheckbox` anywhere, maybe we should remove it and add it back if we need it. Thoughts?